iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -99,17 +98,17 @@ inline typename std::enable_if<!std::is_void<U>::value, const U&>::type Relative
 }
 
 template <typename T>
-inline T* RelativePointer<T>::operator->() const noexcept
+inline typename RelativePointer<T>::ptr_t RelativePointer<T>::operator->() const noexcept
 {
-    auto* ptr = get();
+    auto* const ptr{get()};
     cxx::Ensures(ptr != nullptr);
     return ptr;
 }
 
 template <typename T>
-inline T* RelativePointer<T>::get() const noexcept
+inline typename RelativePointer<T>::ptr_t RelativePointer<T>::get() const noexcept
```

Comment:
I think it would be better to write `T*` as return type. For example https://en.cppreference.com/w/cpp/memory/shared_ptr/get does specify it like this.

Forcing any reader to look up `ptr_t` is annoying and also leads to more code here. I know that it is already necessary in the arguments and it could be argued it should be `T*` there everywhere as well. At some point it only existed to distinguish from `const_ptr_t`, but this was already redundant before and is now.

Reply:
Agreed, I changed it. However, I would keep the alias `ptr_t` for users to find out the type of a `RelativePointer`.